tests/ops/api/v1/endpoints/test_messaging_endpoints.py

```diff
@@ -0,0 +1,761 @@
+        )
+
+    def test_post_email_config_with_no_key(
+        self,
+        db: Session,
+        api_client: TestClient,
+        payload,
+        url,
+        generate_auth_header,
+    ):
+        auth_header = generate_auth_header([MESSAGING_CREATE_OR_UPDATE])
+        response = api_client.post(url, headers=auth_header, json=payload)
+
+        assert 200 == response.status_code
+        response_body = json.loads(response.text)
+
+        assert response_body["key"] == "mailgun"
+        email_config = db.query(MessagingConfig).filter_by(key="mailgun")[0]
+        email_config.delete(db)
+
+    def test_post_email_config_with_invalid_key(
+        self,
+        db: Session,
+        api_client: TestClient,
+        payload,
+        url,
+        generate_auth_header,
+    ):
+        payload["key"] = "*invalid-key"
+        auth_header = generate_auth_header([MESSAGING_CREATE_OR_UPDATE])
+        response = api_client.post(url, headers=auth_header, json=payload)
+        assert 422 == response.status_code
+        assert (
+            json.loads(response.text)["detail"][0]["msg"]
+            == "FidesKey must only contain alphanumeric characters, '.', '_' or '-'."
+        )
+
+    def test_post_email_config_with_key(
+        self,
+        db: Session,
+        api_client: TestClient,
+        payload,
+        url,
+        generate_auth_header,
+    ):
+        payload["key"] = "my_mailgun_messaging_config"
+        auth_header = generate_auth_header([MESSAGING_CREATE_OR_UPDATE])
+
+        response = api_client.post(url, headers=auth_header, json=payload)
+        assert 200 == response.status_code
+
+        response_body = json.loads(response.text)
+        email_config = db.query(MessagingConfig).filter_by(
+            key="my_mailgun_messaging_config"
+        )[0]
+
+        expected_response = {
+            "key": "my_mailgun_messaging_config",
+            "name": "mailgun",
+            "service_type": MessagingServiceType.MAILGUN.value,
+            "details": {
+                MessagingServiceDetails.API_VERSION.value: "v3",
+                MessagingServiceDetails.DOMAIN.value: "my.mailgun.domain",
+                MessagingServiceDetails.IS_EU_DOMAIN.value: False,
+            },
+        }
+        assert expected_response == response_body
+        email_config.delete(db)
+
+    def test_post_email_config_missing_detail(
+        self,
+        api_client: TestClient,
+        url,
+        generate_auth_header,
+    ):
+        auth_header = generate_auth_header([MESSAGING_CREATE_OR_UPDATE])
+        response = api_client.post(
+            url,
+            headers=auth_header,
+            json={
+                "key": "my_mailgun_messaging_config",
+                "name": "mailgun",
+                "service_type": MessagingServiceType.MAILGUN.value,
+                "details": {
+                    # "domain": "my.mailgun.domain"
```

Comment:
Should this be a comment?

Reply:
This was just to show what we purposefully aren't adding, but can be removed